example/test.py

```diff
@@ -35,30 +36,53 @@ def filter_ldconfig_process(ps_rows):
             if not (row.ruser == 'django' and 'ldconfig' in row.args)]
 
 
-def assert_tini_pid_1(ps_row, cmd):
-    args = 'tini -- django-entrypoint.sh {}'.format(cmd)
-    assert_that(ps_row,
-                MatchesStructure.byEquality(pid='1', ruser='root', args=args))
-
-
-def matches_attributes_values(attributes, values):
-    """
-    Returns a matcher that matches the values of several attributes of an
-    object.
-    """
-    return MatchesStructure.byEquality(
-        **{a: v for a, v in zip(attributes, values)})
-
-
-def matches_ruser_args_unordered(*expected_values):
-    """
-    Returns a matcher that, given a list of PsRow objects, ensures that objects
-    are present that have 'ruser' and 'args' values that match those given.
-    """
-    def row_matcher(values):
-        return matches_attributes_values(['ruser', 'args'], values)
-
-    return MatchesSetwise(*map(row_matcher, expected_values))
+class PsTreeMismatch(Mismatch):
+    def __init__(self, row_fields, child_count, fields_mm, children_mm):
+        self.row_fields = row_fields
+        self.child_count = child_count
+        self.fields_mm = fields_mm
+        self.children_mm = children_mm
+
+    def describe(self):
+        rfs = ['{}={!r}'.format(k, v)
+               for k, v in sorted(self.row_fields.items())]
+        suffix = '' if self.child_count == 1 else 'ren'
+        descriptions = ['PsTree({} with {} child{}) mismatch: ['.format(
+            ', '.join(rfs), self.child_count, suffix)]
+        if self.fields_mm is not None:
+            for m in self.fields_mm.mismatches:
+                for l in m.describe().splitlines():
+                    descriptions.append('  ' + l.rstrip('\n'))
+        if self.children_mm is not None:
+            descriptions.append('  mismatches in children:')
+            for l in self.children_mm.describe().splitlines():
+                descriptions.append('    ' + l.rstrip('\n'))
+        descriptions.append(']')
+        return '\n'.join(descriptions)
+
+
+class MatchesPsTree(object):
+    def __init__(self, ruser, args, ppid=None, pid=None, children=()):
```

Comment:
For completeness, I think it does. There's no reason *not* to have it, and people might want to examine subtrees and assert that the subtree's root has the right `ppid`.